crates/utils/src/error.rs

```diff
@@ -171,16 +147,46 @@ pub enum LemmyErrorType {
   /// Thrown when an API call is submitted with more than 1000 array elements, see
   /// [[MAX_API_PARAM_ELEMENTS]]
   TooManyItems,
-  CommunityHasNoFollowers,
   BanExpirationInPast,
   InvalidUnixTime,
   InvalidBotAction,
   CantBlockLocalInstance,
+  Unknown(String),
+  UrlLengthOverflow,
+  FederationError(Option<FederationError>),
+}
+
+/// Federation related errors, these dont need to be translated.
+#[derive(Display, Debug, Serialize, Deserialize, Clone, PartialEq, Eq, EnumIter, Hash)]
+#[cfg_attr(feature = "full", derive(ts_rs::TS))]
+#[cfg_attr(feature = "full", ts(export))]
+#[serde(tag = "error", content = "message", rename_all = "snake_case")]
+#[non_exhaustive]
+pub enum FederationError {
```

Comment:
I might be way off, but should this live in the activitypub federation crate? A lot of these seem lemmy specific so probably not.

Reply:
That would make sense, but as you can see its used in LemmyErrorType. If its defined in another crate thats not possible, and Im not sure how it could be handled then.

Reply:
Cool, no big deal.